tools/teamcity-generator/main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+var GA_VERSION = "ga"
+var BETA_VERSION = "beta"
+
+var oPath = flag.String("output", "", "path to output generated files to")
+var ver = flag.String("version", "", "version name, value must be `ga` or `beta`")
+
+func main() {
+	var version string
+	var outputPath string
+
+	flag.Parse()
+	outputPath = *oPath
+	version = *ver
+	if outputPath == "" {
+		log.Fatalf("missing output flag: provide `--output <path>` to set the path to output generated files to")
+	}
+	if version == "" {
+		log.Fatalf("missing version flag: provide `--version <ga|beta>` to set the path to output generated files to")
+	}
+	if version != GA_VERSION && version != BETA_VERSION {
+		log.Fatalf("invalid version flag value: value must be `%s` or `%s`", GA_VERSION, BETA_VERSION)
+	}
+
+	var terraformResourceDirectory string
+	switch version {
+	case GA_VERSION:
+		terraformResourceDirectory = "google"
+	case BETA_VERSION:
+		terraformResourceDirectory = "google-beta"
+	default:
+		log.Fatalf("invalid version flag value: value must be `%s` or `%s`", GA_VERSION, BETA_VERSION)
+	}
+
+	log.Println("Generating TeamCity configuration service package map for `%s` provider", terraformResourceDirectory)
+
+	// Get a list of the service packages found in a given directory
+	servicesDir := fmt.Sprintf("%s/%s/services", outputPath, terraformResourceDirectory)
+	serviceList, err := readAllServicePackages(servicesDir)
+	if err != nil {
+		log.Fatalf("error determining service package list: %s", err)
+	}
+
+	// Create a string of the map that should be created in .teamcity/components/generated/services.kt
+	serviceMap, err := createMap(serviceList)
+	if err != nil {
+		log.Fatalf("error creating service package map: %s", err)
+	}
+
+	// Ensure .teamcity/components/generated/services.kt exists, create if not present
+	// "Create creates or truncates the named file. If the file already exists, it is truncated."
+	servicesKtFilePath := fmt.Sprintf("%s/.teamcity/components/generated/services.kt", outputPath)
+	log.Printf("Opening %s", servicesKtFilePath)
+	f, err := os.Create(servicesKtFilePath)
```

Comment:
The file will exist, as MMv1 is still generating it due to work in this PR: https://github.com/GoogleCloudPlatform/magic-modules/pull/8254

We could either:
- Remove generation of services.kt from MMv1 and return it later when this tool is deleted (after DCL => MMv1 migration is complete)
- Leave MMv1 to generate a services.kt file that's overwritten in the next generation step by this tool (until it is deleted in future)

Reply:
I think both ways work.

Reply:
Yeah same here. I prefer leaving the two tools both generating the file as it'll make the eventual clean-up easier.

Until then, I hope that the two different comments (`// this file is auto-generated with mmv1...` and `// this file is auto-generated by magic-modules/tools/teamcity-generator...`) help avoid any confusion!